backend/src/router.go

```diff
@@ -349,3 +351,48 @@ func getProfile(c *gin.Context) {
 		"user": users[0],
 	})
 }
+
+
+//not all of this is needed to make an update, at least according to the guide. We shall see.
+type ProfileUpdateData struct { 
+	ID        uint   `json:"-"`
+	FirstName string `json:"firstname"`
+	LastName  string `json:"lastname"`
+	Email     string `json:"email"`
+	Phone     string `json:"phone"`
+}
+
+// ADD DESCRIPTION
+func patchProfile(c *gin.Context) {
+	
+	token, err := c.Cookie("jwt")
+	if err != nil {
+		c.JSON(401, gin.H{
+			"error": "Requires an authenticated user.",
+		})
+		return
+	}
+	
+	claims, err := ParseJWT(token)
+	if err != nil {
+		c.JSON(500, gin.H{
+			"error": "Unable to parse JWT: " + err.Error() + ".",
+		})
+		return
+	}
+	
+	var users []User
+	//add error checking for nonexisting user 
+	//Note for self: https://blog.logrocket.com/how-to-build-a-rest-api-with-golang-using-gin-and-gorm/
+	
+	var edits ProfileUpdateData
+	if err := c.ShouldBindJSON(&edits); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+	
+	DB.Find(&users, "username = ?", claims.Username)
+	DB.Model(&users[0]).Updates(edits)
```

Comment:
I saw the latest commit regarding some issue with Gorm here and took a closer look. [According to the docs](https://gorm.io/docs/update.html#Updates-multiple-columns), `Updates` with structs will not change default values which is what we need (that way a user can remove an email or something). We also need to be careful about distinguishing between default values passed in the body to be updated versus default values applied from binding. Not sure the best way to handle this; it's probably a separate issue.